### ISSUE TITLE
On slow internet, the iyo 10 second timeout is not handled properly

### ISSUE THREAD

**Issue by robvanmieghem** [gigforks/gogs#21]
```
HTTP/2.0 500 Internal Server Error
Content-Type: text/plain; charset=utf-8
Date: Wed, 11 Jan 2017 09:22:49 GMT
Server: Caddy
Set-Cookie: i_like_gogits=d2201309c3bc0e9a; Path=/; HttpOnly
_csrf=fTcFOhgBUF-C_tnUJExU-JejmjY6MTQ4NDEyNjU2OTA0MzUxOTcwMw%3D%3D; Path=/; Expires=Thu, 12 Jan 2017 09:22:49 GMT; HttpOnly
Content-Length: 0
X-Firefox-Spdy: h2
```

**Comment by arahmanhamdy** [comment]
Needs more clarification. Anyway if it is related to signin, the flow already has been changed in gitea.

**Comment by LeeSmet** [comment]
Once a user logs in through the oauth flow in IYO, the gogs server receives a callback with a code. This code can then be exchanged for an access token. However, there is only a 10 second window where this is possible. On slow internet, the request may exceed this deadline, and IYO returns an error. The original integration could not handle this, thus returning an internal server error. 

As far as I understand, you are now using a library for standard oauth2, so if the library is any decent, it should handle this

**Comment by Dinaamagdy** [comment]
solved on gitea 1.0.0 at f76784f3c51474c9dca3e5bf3ed3b5a1a3baba2e
